### PR TITLE
Add sql sanitizer

### PR DESCRIPTION
**Pull request by josh-lauer**
Pivotal story: https://www.pivotaltracker.com/story/show/169434862

Adds SQL sanitization

### CODE REVIEW

File: lib/active_record/open_tracing/processor.rb [hunk 1]
```diff
@@ -45,12 +48,16 @@ def tags_for_payload(payload)
...
           "db.cached" => payload.fetch(:cached, false),
-          "db.statement" => payload.fetch(:sql).squish,
+          "db.statement" => sanitize_sql(payload.fetch(:sql).squish),
           "db.type" => DB_TYPE,
           "peer.address" => db_address
         }
       end
 
+      def sanitize_sql(sql)
+        sanitizer ? sanitizer.sanitize(sql) : sql
```
Comment (jeremiahishere) [inline lib/active_record/open_tracing/processor.rb]: should this return empty string if the sanitizer is not defined?

Comment (josh-lauer) [reply lib/active_record/open_tracing/processor.rb]: That's probably a safer default behavior. I will change that.

Comment (josh-lauer) [reply lib/active_record/open_tracing/processor.rb]: I added an extra configuration flag to optionally disable SQL logging entirely. In `dox-tracing` I will set the sanitizer to default to `:mysql`.

**Review by jeremiahishere — APPROVED** [review]
🍤

**Pull request merged**